src/components/ThemeSwitcher.tsx

```diff
@@ -0,0 +1,73 @@
+import { Flex, Image, Text, ButtonGroup, Button } from "@mantine/core"
+import { useAtom } from "jotai"
+import { Icon } from "@iconify/react"
+import cx from "classnames"
+import { useMutation } from "@tanstack/react-query"
+
+import { $theme } from "../store/theme"
+
+import { THEMES } from "../themes"
+import { loginToSite } from "../utils/switch-site"
+import { ThemeKey } from "../types/theme"
+
+export const ThemeSwitcher = () => {
+  const [activeTheme, setActiveTheme] = useAtom($theme)
+
+  const loginMutation = useMutation({
```

Comment:
This handles the loading state for us. We show a loading indicator on the site change button, so the user has an immediate feedback.